[CPF-3147] EasyDAO-TransactionDAO

jlhughes opened the pull request.

jlhughes committed:
Log rule id on exception to help in troubleshooting

jlhughes committed:
When oldObj null follow 'new' logic path.

lchanmann commented on src/foam/nanos/ruler/predicate/PropertyEQValue.js:
```diff
@@ -40,11 +40,12 @@ foam.CLASS({
-        if ( getIsNew() ) {
+        FObject old = (FObject) OLD_OBJ.f(obj);
+        if ( getIsNew() ||
+             old == null ) {
```
when isNew=false, we want to compare the oldObj.prop to the value and if the oldObj is null should we just return false? @nanoMichal 

(similar thing for ropertyNEQValue.js and PropertyEQProperty.js)

jlhughes replied:
without the check for old == null we have NPE because the old tests are blindly executed.

lchanmann replied:
we can add the null check before the last return statement.
```
return old != null && EQ(old.getClassInfo().getAxiomByName ...
```

Commit:
Make delete modal buttons consistent with rest of app

Custom CSS should almost never be used for buttons anymore. Instead, the
style of buttons is derived from a few properties on ActionView.

lchanmann commented on src/foam/nanos/ruler/Rule.js:
```diff
@@ -275,9 +275,8 @@
-
-          ((Logger) x.get("logger")).error(
-            "Failed to evaluate predicate of rule: " + getId(), t);
+          // ((Logger) x.get("logger")).debug(this.getClass().getSimpleName(), "id", getId(), "\\nrule", this, "\\nobj", obj, "\\nold", oldObj, "\\n", t);
```
do we want to leave the commented code here?

jlhughes replied:
yes. debugging rules is tricky, and keep enabling these to see what is going on.

lchanmann commented on src/foam/nanos/ruler/RuleEngine.java:
```diff
@@ -57,21 +57,27 @@ public void setDelegate(DAO delegate) {
+        pm.log(x_);
+        agency.submit(x_, x -> saveHistory(rule, obj), "Save history. Rule id:" + rule.getId());
+      } catch ( Exception e ) {
+        // logger.debug(this.getClass().getSimpleName(), "id", rule.getId(), "\\nrule", rule, "\\nobj", obj, "\\nold", oldObj, "\\n", e);
```
same as above

jlhughes committed:
Merge pull request #2919 from foam-framework/CPF-3839/fix-delete-modal-button-styles

Make delete modal buttons consistent with rest of app

jlhughes committed:
Merge branch 'master' of github.com:foam-framework/foam2 into CPF-3147/EasyDAO-TransactionDAO

jlhughes committed:
Return false when old object is null rather than default to testing new object.
